### ISSUE TITLE
@cdn is undocumented

### ISSUE THREAD

**Issue by lpsinger** [architect/architect#1483]
**Describe the issue**
The `@cdn` pragma is undocumented.

**Steps to reproduce**
N/A

**Expected behavior**
It should be documented at https://arc.codes.

**Screenshots**
N/A

**Desktop**
N/A

**Mobile**
N/A

**Additional context**
N/A

**Comment by ryanblock** [comment]
My suggestion as the fastest workaround here would just be to let it perform its invalidation, if possible; apparently that means ensuring the role has `cloudfront:CreateInvalidation` (although ideally it's `AdministratorAccess`, per our setup requirements: https://arc.codes/docs/en/get-started/detailed-aws-setup#credentials).

**Comment by lpsinger** [comment]
It's lucky that we don't deploy with `AdministratorAccess`, because when I added the following to our app.arc, Architect tried to delete our CloudFormation distribution!

```
@cdn
false
```

```
⚬ Deploy Destroying API Gateway CDN distribution
AccessDenied: @aws-lite/client: CloudFront.GetDistributionConfig: User: arn:aws:sts::.../GitHubActions is not authorized to perform: cloudfront:GetDistributionConfig on resource: arn:aws:cloudfront::... because no identity-based policy allows the cloudfront:GetDistributionConfig action
```

**Comment by ryanblock** [comment]
That's not correct. When you use `@cdn false` Arc will actively attempt to clean up any CDN-related resources; in this case, it's attempting to [clean up the distribution configuration for your API Gateway](https://github.com/architect/deploy/blob/202f8404618bffcf7d4ae90562aaf51be8d921a7/src/sam/02-after/00-get-app-apex/index.js) – not your CloudFormation distribution. Remove `@cdn false` and proceed with `AdministratorAccess` and you'll be fine!

**Comment by lpsinger** [comment]
> Remove @cdn false and proceed with AdministratorAccess and you'll be fine! 

Our cloud engineering team definitely will not allow us to run the CI/CD pipeline with AdministratorAccess.

**Comment by ryanblock** [comment]
As mentioned before, you can also go ahead and add `cloudfront:CreateInvalidation` to your permissions and you should be fine.

**Comment by lpsinger** [comment]
But we don't want Architect to invalidate our CDN. We fingerprint our assets.

**Comment by ryanblock** [comment]
The thinking behind the invalidation is that while assets may be fingerprinted, you may have some amount of caching on your HTML / JSON etc. This has proven to be quite useful in a few instances.

However, I recognize that it's still a bit tricky: we aren't prioritizing actually (re)building this feature right, and I don't want to rip it out because there are a handful of users who've incorporated it into their workflows, and I don't want to introduce a breaking change for something that's denoted as undocumented and experimental. Of course we also don't want to hold you back here, so I'm open to ideas.

**Comment by lpsinger** [comment]
If it's experimental, then could you add a way to disable it?

**Comment by DocLM** [comment]
Hi everyone, I propose to handle invalidations only if CDN is explicity enabled.

Look to https://github.com/architect/deploy/pull/389 for changes.

**Comment by DocLM** [comment]
In general, probably you should filter distributions list in https://github.com/architect/deploy/blob/v5.0.7/src/sam/02-after/00-get-app-apex/index.js to CDN managed by Architect.

Also the current approach in https://github.com/architect/deploy/blob/v5.0.7/src/sam/02-after/00-get-app-apex/reads.js is looking for distributions that are only in the same account and matching the **first** origin with the API Gateway or S3 bucket created by Architect.

I came across this issue because Architect tried to invalidate a distribution that merge together multiple Architect managed apps but casually the first origin matchs one of the apps.

**Comment by lpsinger** [comment]
@ryanblock, I'd like to get back to tracking recent release of Architect, but this issue presents a significant performance regression for us. Would you please review https://github.com/architect/deploy/pull/389?
